myia/prim/shape_inferrers.py

```diff
@@ -0,0 +1,161 @@
+"""Definition of shape inference for primitives."""
+
+import operator
+from functools import partial, reduce
+
+from ..infer import ANYTHING, GraphInferrer, register_inferrer, \
+    Track, MyiaShapeError, Inferrer
+from ..ir import Graph
+from ..dtype import Array
+
+from . import ops as P
+from .ops import Primitive
+
+
+def prod(iterable):
+    """Return the product of the elements of the iterator."""
+    return reduce(operator.mul, iterable, 1)
+
+
+shape_inferrer_constructors = {}
```

Comment:
I think I missed when this gets populated.

Reply:
From the `@shape_inferrer` decorator, which is a partial on `register_inferrer` and this dictionary.